packages/eslint-plugin/src/rules/no-explicit-any.ts

```diff
@@ -21,6 +22,9 @@ export default util.createRule({
         context.report({
           node,
           messageId: 'unexpectedAny',
+          fix(fixer) {
```

Comment:
could you please hide the fixer behind a rule option?
I'd suggest something along the lines of `fixToUnknown: boolean`, with the default set to false - on the next major version bump, we can change it to `true` by default.
Don't forget to document the option in the rule docs!

### Why?

ESLint fixers shouldn't break the code.

[From the docs:](https://eslint.org/docs/developer-guide/working-with-rules#applying-fixes)
> Avoid any fixes that could change the runtime behavior of code and cause it to stop working.

Whilst this doesn't technically change the runtime behaviour of the code, it could cause compiler errors, breaking the codebase.

I.e.
```
const x: any = 1;
const y = x + 1;
```
changing the `any` to `unknown` here will cause compiler errors.

Gating it behind an option allows people to chose not to have the potentially breaking fixer.

Reply:
I think it doesn't automatically fix the code while reporting it just fix the problems when we run eslint --fix in CLI

Reply:
That is correct, but a lot of people use eslint with autofix on save, or they do things like blindly running fix over their codebase.

In either case right now that will do nothing for this rule. So if we land this PR without a breaking version bump, it could cause problems for users because they aren't expecting it.

Reply:
But I didn't find any commands for the fixToUnknown: boolean, in the EsLint docs.

Reply:
sorry, I meant add a new option entirely.

line 15 there is an empty array for `schema`.
Inside here you need to place a JSON schema which defines an object with one property, `fixToUnknown`.

line 18 there is an empty array for `defaultOptions`.
This should become `[{ fixToUnknown: false }]`.

You'll need to add types for the options, which will be passed in as a generic into the `util.createRule` call.

the `create` function accepts a second argument, being the options with defaults applied, so line 19 can become `create(context, [{ fixToUnknown }]) {`

Then within your fix function, you can consume the new option and only do the fix if the option is true.

Reply:
Wow Thank You for the detailed description